src/functions/ducklake_get_option.cpp

```diff
@@ -0,0 +1,284 @@
+#include "functions/ducklake_table_functions.hpp"
+#include "storage/ducklake_transaction.hpp"
+#include "storage/ducklake_catalog.hpp"
+#include "storage/ducklake_schema_entry.hpp"
+#include "storage/ducklake_table_entry.hpp"
+#include <unordered_map>
+
+namespace duckdb {
+
+struct OptionMetadata {
+	string description;
+	string type;
+};
+
+static constexpr const char *OPTION_TYPE_USER = "user";
+static constexpr const char *OPTION_TYPE_SYSTEM = "system";
+static constexpr const char *DEFAULT_OPTION_DESCRIPTION = "Custom configuration option";
+
+static constexpr const char *COLUMN_NAME = "name";
```

Comment:
This being static isn't done elsewhere in the code - perhaps we can just inline it for consistency?